### ISSUE TITLE
GBDT::BoostFromAverage() question

### ISSUE THREAD

**Issue by IbanezJS** [Microsoft/LightGBM#1546]
Hi 

`for (auto& score_updater : valid_score_updater_) {
          score_updater->AddScore(init_score, 0);
        }`

Why you use init_score from train dataset labels for validation score updaters?
Each validation data has its own initial score.

**Comment by guolinke** [comment]
@IbanezJS 
There valid data actually are testing data. We assume we don't know their labels.

**Comment by IbanezJS** [comment]
@guolinke 
kinda doubtfully

**Comment by guolinke** [comment]
@IbanezJS 
It is the basic concept of machine learning. We use the training data to learn the model, and use the model for the unseen data. 
Valid data is the simulation of testing data, so we cannot use their labels, as we don't know the label of testing data.
